Add Available for Download Date

little9 opened the pull request:
This changes the localization for 'Embargo release date' to 'Available for Download Date', updates the `Etd` factory with an embargo and adds a view spec for testing that the release date is displaying on the show page. It also removes this attribute from the search results.

little9 committed:
Changes Embargo release to 'Available for Download Date'

Updates the `Etd` factory with an embargo and adds a view spec for testing that
the release date is displaying on the show page. It also removes this attribute
from the search results.

little9 commented:
@no-reply I'm not sure why I'm getting locale errors on travis, they aren't happening for me locally.

no-reply commented on spec/factories/etds.rb:
```diff
@@ -51,6 +51,7 @@
       rights_note      ['For the exclusive viewing of Little My.',
                         'Moomin: do not read this.']
       rights_statement [RDF::URI('http://rightsstatements.org/vocab/NKC/1.0/')]
+      embargo          Hydra::AccessControls::Embargo.new(id: 'test-embargo', visibility_during_embargo: false, visibility_after_embargo: true, embargo_release_date: 'January 1, 3000')
```
I think we should not make `moomins_thesis` embargoed. This might have an impact on several other tests that would require a generally populated thesis.

Also, this factory is causing me problems running specs locally. After running the suite once, I get `Attempting to recreate existing ldp_source: `http://127.0.0.1:8986/rest/test/te/st/-e/mb/test-embargo' on subsequent runs.

no-reply commented on spec/views/hyrax/base/_metadata.html.erb_spec.rb:
```diff
@@ -0,0 +1,16 @@
+  let(:solr_document) { SolrDocument.new(work.to_solr) }
+  let(:work)          { FactoryGirl.build(:moomins_thesis) }
+
+  it { is_expected.to match('Available for Download Date') }
```
Could we use the `have_show_field` matcher here?

no-reply commented on app/controllers/catalog_controller.rb:
```diff
@@ -87,7 +87,6 @@ def self.modified_field
     config.add_index_field solr_name("date_created", :stored_searchable), itemprop: 'dateCreated'
     config.add_index_field solr_name("resource_type", :stored_searchable), label: "Document Type", link_to_search: solr_name("resource_type", :facetable)
     config.add_index_field solr_name("file_format", :stored_searchable), link_to_search: solr_name("file_format", :facetable)
-    config.add_index_field solr_name("embargo_release_date", :stored_sortable, type: :date), label: "Embargo release date", helper_method: :human_readable_date
```
Can you comment about why we are dropping this line from the catalog controller? It's not clear to me from the ticket or commit message what this change accomplishes.

little9 replied:
One of the acceptance criteria is that the release date doesn't show up in the search results. 

This removes it from the
```ruby
# solr fields to be displayed in the index (search results) view
#   The ordering of the field names is the order of the display
```
section in the catalog controller.

no-reply replied:
Let's test this in https://github.com/curationexperts/mahonia/blob/master/spec/views/catalog/_index_list_default.html.erb_spec.rb#L21 ?

no-reply reviewed (changes requested):
I have a couple of questions, and some suggested changes that I'm working on on my end.

Commit:
Move the license spec to the metadata partial view specs

We had a standing `@todo` item to figure out a better place to test the
license. Now that we have `_metadata` partial specs setup, it's better to test
this there.

Commit:
Move all `_attribute_rows` specs to `_metadata`

The idea behind this is that we don't actually care about the `_attribute_rows`
partial, and would be indifferent to see it squashed into `_metadata`. We can
test all the metadata for the show view in one place.

The downsides of this are: there isn't a spec file corresponding directly with
the `_attribute_rows.html.erb` partial, which we customize; hyrax's test suite
tests `_attribute_rows`, so this is a minor divergence from them.

Commit:
Test the embargo ttl export to `Hydra::ACL::hasEmbargo`

little9 reviewed (commented):
👍
